scala3doc/resources/dotty_res/scripts/components/DocumentableList.js

```diff
@@ -68,83 +57,125 @@ class DocumentableList extends Component {
   }
 }
 
-class List {
-  filterTab(name) {
-    return name !== "Linear supertypes" && name !== "Known subtypes" && name !== "Type hierarchy"
-  }
-
+class List { 
+  /**
+   * @param tabsRef { Element[] }
+   * @param sectionRefs { Element[] }
+   */
   constructor(tabsRef, sectionRefs) {
     this._tabsRef = tabsRef;
     this._sectionRefs = sectionRefs;
   }
 
   get tabsRefs() {
-    return this._tabsRef.filter((tabRef) => this.filterTab(this._getTogglable(tabRef)));
+    return this._tabsRef.filter(tabRef => this.filterTab(this._getTogglable(tabRef)));
   }
 
   get sectionsRefs() {
-    return this._sectionRefs.filter( (sectionRef) => this.filterTab(this._getTogglable(sectionRef)));
+    return this._sectionRefs.filter(sectionRef => this.filterTab(this._getTogglable(sectionRef)));
+  }
+
+  /**
+  * @param name { string }
+  */
+  filterTab(name) { 
+    return name !== "Linear supertypes" && name !== "Known subtypes" && name !== "Type hierarchy"
   }
 
+  /**
+   * @param sectionRef { Element }
+   */
   getTabRefFromSectionRef(sectionRef) {
     return this.tabsRefs.find(
       (tabRef) => this._getTogglable(tabRef) === this._getTogglable(sectionRef)
     );
   }
 
+  /**
+   * @param sectionRef { Element }
+   * @returns { Element[] }
+   */
   getSectionListRefs(sectionRef) {
     return findRefs(".documentableList", sectionRef);
   }
 
+  /**
+   * @param listRef { Element }
+   * @returns { Element[] }
+   */
   getSectionListElementsRefs(listRef) {
     return findRefs(".documentableElement", listRef);
   }
 
-  mapListElementRef(elementRef) {
+  /**
+   * @param elementRef { Element }
+   * @returns { ListElement }
+   */
+  toListElement(elementRef) {
     return {
       ref: elementRef,
       name: getElementTextContent(getElementNameRef(elementRef)),
       description: getElementTextContent(getElementDescription(elementRef)),
     };
   }
 
+  /**
+  * @param elementData { ListElement }
+  * @param filter { Filter }
+  */
   isElementVisible(elementData, filter) {
-    if (!this._areFiltersFromElementSelected(elementData, filter)) {
-      return false;
+    return !areFiltersFromElementSelected() 
```

Comment:
nitpicking: I'm not a fan of negative ternary conditions when unnecessary - Otherwise LGTM :smiley:

Reply:
I'm planning on streamlining the logic after adding some integration tests - this is more complicated that it should be, not only here, but for now - it works. ^^

Reply:
ok! if it works now and will be streamlined later then all good, we can merge as soon as the tests finish